vulcanai2/models/basenetwork.py

```diff
@@ -434,30 +434,51 @@ def _validate(self, val_loader):
 
         return validation_loss, validation_accuracy
 
-    def run_test(self, test_x, test_y, figure_path=None, plot=False):
+    def run_test(self, data_loader, figure_path=None, plot=False):
         """
         Will conduct the test suite to determine model strength.
         """
-        return self.metrics.run_test(self, test_x, test_y, figure_path, plot)
+        return self.metrics.run_test(
+            network=self,
+            data_loader=data_loader,
+            figure_path=figure_path,
+            plot=plot)
 
     # TODO: Instead of self.cpu(), use is_cuda to know if you can use gpu
-    def forward_pass(self, input_data, convert_to_class=False):
+    def forward_pass(self, data_loader, convert_to_class=False):
         """
         Allow the implementer to quickly get outputs from the network.
 
-        Args:
-            input_data: Numpy matrix to make the predictions on
-            convert_to_class: If true, output the class
-                             with highest probability
+        :param data_loader: DataLoader object to make the predictions on
+        :param convert_to_class: If true, list of class predictions instead
+                                 of class probabilites
 
-        Returns: Numpy matrix with the output probabilities
-                 with each class unless otherwise specified.
+        :return: Numpy matrix with the output probabilities
+                 for each class unless otherwise specified.
         """
-        output = self.cpu()(torch.Tensor(input_data)).data.numpy()
-        if convert_to_class:
-            return self.metrics.get_class(output)
-        else:
-            return output
+        self.eval()
+        # prediction_shape used to aggregate network outputs
+        # (e.g. with or without class conversion)
+        pred_collector = torch.tensor([])
+        for batch_idx, (data, _) in enumerate(data_loader):
+            if torch.cuda.is_available():
+                data = data.cuda()
+            # Get raw network output
+            predictions = self(data)
+            if self._num_classes:
+                # Get probabilities
+                predictions = nn.Softmax(dim=1)(predictions)
+                if convert_to_class:
+                    predictions = torch.tensor(
+                        self.metrics.get_class(in_matrix=predictions)).float()
+            # Aggregate predictions
+            pred_collector = torch.cat([pred_collector, predictions])
```

Comment:
Change to: `pred_collector = torch.cat([pred_collector, predictions.cpu()])`

Reply:
will do